c++/src/kj/compat/http.c++

```diff
@@ -1952,12 +1952,16 @@ public:
       // counter immediately, so that we still know where we are in case of cancellation.
       auto amount = co_await getInner().tryRead(buffer, 1, kj::min(maxBytes, length));
 
-      length -= amount;
-      if (length > 0) {
+      if (length > alreadyRead) {
         // We haven't reached the end of the entity body yet.
         if (amount == 0) {
-          kj::throwRecoverableException(KJ_EXCEPTION(DISCONNECTED,
-              "premature EOF in HTTP entity body; did not reach Content-Length"));
+          kj::throwRecoverableException(KJ_EXCEPTION(
+            DISCONNECTED,
+            "premature EOF in HTTP entity body; did not reach Content-Length. Expected length: ",
```

Comment:
KJ exceptions do not simply concatenate the parameters, they produce a formatted error string that includes both the variable names and values. Your exception will end up looking like this:

`premature EOF in HTTP entity body; did not reach Content-Length. Expected length: ; length = 123; Received length: ; alreadyRead = 12`

I would suggest just doing: `KJ_EXCEPTION("premature EOF in HTTP entity body; did not reach Content-Length", length, alreadyRead)`

Reply:
Thanks! Switched to that way of formatting the exception!